app/server.js

```diff
@@ -54,3 +57,63 @@ controller.on(['ambient', 'direct_message', 'file_share'], (bot, message) => {
 controller.hears(['show'], ['direct_message'], (bot, message) => {
   console.log(channelMessages)
 })
+
+/************************* milestones *************************/
+
+var channels = ['C9G17060H', 'C9G073P2N']   // dummy channel list
+
+// start and end dates for term and week count
+var startDate = moment('03/28/2018', 'MM/DD/YYYY')
+var endDate = moment('05/30/2018', 'MM/DD/YYYY')
+var currentWeek = 0
+
+// schedule milestone message for Wednesday after the Lab meeting at 8pm
+var rule = new schedule.RecurrenceRule() 
+rule.dayOfWeek = 3
+rule.hour = 20
+rule.minute = 0
+rule.second = 0 
+
+// if the current date is within the start and end date range
+if (moment().isSameOrAfter(startDate) && moment().isSameOrBefore(endDate) && currentWeek <= 8) {
+  
+  // post milestones message in each channel in channels list if it's the corresponding time
+  var milestones = schedule.scheduleJob(rule, function(){
+    for (var i = 0; i < channels.length;  i++) {
+      slackbot.say( {
+        text: getMilestone(currentWeek),
+        channel: channels[i]
+      }) 
+    }
+    currentWeek++
+  }) 
+}
+
+/*
+* getMilestone() takes in the current week number and gets and formats the milestone for
+* that week.
+*/
+function getMilestone(week) {
+  var raw_milestone = data[week] // pulls raw milestone
+  var milestone = '@channel _It\'s milestone time!_\n\n' 
+
+  // adds weekly milestone title and week #
+  milestone += '*Week ' + raw_milestone['milestone'] + ':* ' + (raw_milestone['title'] + '\n')
+
+  // formats each section of the milestone in markdown format
+  for (var section in raw_milestone) {
+    if (section != 'milestone' && section != 'title') {
+      milestone += markdown.formatHeader(section) 
+      if (typeof raw_milestone[section] === typeof new Object()) {
+        milestone += markdown.formatLists(raw_milestone[section]) 
+      } else {
+        milestone += (raw_milestone[section] + '\n')
+      }
+    }
+  }
+
+  // tacks on the 'But wait, there's more!' statement
+  milestone += "\nBut wait, there's more! Check out https://build.dali.dartmouth.edu for the full list of tasks." 
+
+  return milestone
+}
```

Comment:
Make sure that all your files end with a new line. Try to find a plugin (or possible change VSCode's settings) to automatically insert final newline characters on save.